File: special-positions-in-a-binary-matrix.ts
{
    const numSpecial = function(mat) {
        let sum = 0;

        for(let i = 0; i < mat.length; i++) {
            
            for(let j = 0; j < mat[0].length; j++) {
                const col = mat.map(x => x[j]).filter(x => x == 1)

                const row = mat[i].filter(x => x == 1)

                if (mat[i][j] == 1 && row.length == 1 && col.length == 1) 
                    sum += 1
            }

        }

        return sum
    };
}

File: special-positions-in-a-binary-matrix.rs
pub fn num_special(mat: Vec<Vec<i32>>) -> i32 {
    let m = mat.len();
    let n = mat[0].len();
    let mut count = 0;

    for i in 0..m {
        for j in 0..n {
            if mat[i][j] == 1 {
                let row_sum: i32 = mat[i].iter().sum();
                let col_sum: i32 = (0..m).map(|r| mat[r][j]).sum();
                if row_sum == 1 && col_sum == 1 {
                    count += 1;
                }
            }
        }
    }

    count
}
